samples/matlab_experimental/test_examples.m

```diff
@@ -1,31 +1,52 @@
-% runs selected examples without pausing
-
-ctLoad
-clear all
-close all
-
-equil();
-isentropic();
-reactor1();
-reactor2();
-surf_reactor;
-periodic_cstr;
-% plug_flow_reactor;
-lithium_ion_battery
-rankine;
-prandtl1();
-prandtl2();
-% flame1;
-flame2;
-catcomb;
-diff_flame;
-ignite;
-ignite_hp;
-ignite_uv;
-diamond_cvd;
-
-clear all
-close all
-ctUnload
-
-disp('Test example run successful.')
+function success = test_examples()
+    ctLoad
+
+    examples = {
+        'equil', 'isentropic', 'reactor1', 'reactor2', 'surf_reactor', ...
+        'periodic_cstr', 'plug_flow_reactor', 'lithium_ion_battery', ...
+        'rankine', 'prandtl1', 'prandtl2', 'catcomb','ignite', 'diamond_cvd', ...
+        % 'flame1', 'flame2', 'diff_flame', 'ignite_hp', 'ignite_uv'
+    };
+
+    success = true;
+    passed = {};
+    failed = {};
+
+    for idx = 1:length(examples)
+        scriptName = examples{idx};
+        try
+            evalin('base', sprintf('run(''%s.m'')', scriptName));
+            passed{end+1} = scriptName;
+        catch ME
+            fprintf('An error occurred while running %s: %s\n', scriptName, ME.message);
+            fprintf('Identifier: %s\n', ME.identifier);
+            if strcmp(ME.identifier, 'Cantera:ctError')
+                disp('Caught a CanteraError. Continuing execution...\n');
+            end
+            failed{end+1} = scriptName;
+            success = false;
+        end
+    end
+
+
+    % Summary report
+    disp(' ');
+    disp('============================');
+    disp('Summary of example runs');
+    disp('============================');
+
+    if ~isempty(passed)
+        fprintf('✅ Passed: %s\n', strjoin(passed, ', '));
+    else
+        disp('✅ Passed: (none)');
+    end
+
+    if ~isempty(failed)
+        fprintf('❌ Failed: %s\n', strjoin(failed, ', '));
+    else
+        disp('❌ Failed: (none)');
+    end
+    ctUnload
+
+    disp('Test example run complete.');
+end
```

Comment:
Please add a blank newline to the end of this file. Note the message from GitHub in the diff view.